src/client.ts

```diff
@@ -72,7 +62,9 @@ export class SessionBackend {
           console.error(`Error parsing status stream message as JSON: "${text}"`, e)
         }
         if (data?.state === 'Ready') {
-          this.openSocket()
+          this._isReady = true
+          this._onReady.forEach((cb) => cb())
+          this._onReady = []
```

Comment:
I think a rebase/merge will remove these changes from this PR (since this has already landed in `main`). Let me know if you run into any problems with this!